Sofa/Component/Topology/Mapping/src/sofa/component/topology/mapping/Tetra2TriangleTopologicalMapping.cpp

```diff
@@ -65,38 +65,54 @@ Tetra2TriangleTopologicalMapping::Tetra2TriangleTopologicalMapping()
 void Tetra2TriangleTopologicalMapping::init()
 {
     bool modelsOk = true;
+
+    // Check input topology
     if (!fromModel)
     {
         // If the input topology link isn't set by the user, the TopologicalMapping::create method tries to find it.
         // If it is null at this point, it means no input mesh topology could be found.
         msg_error() << "No input mesh topology found. Consider setting the '" << fromModel.getName() << "' data attribute.";
         modelsOk = false;
     }
+    else
+    {
+        // Making sure the input topology corresponds to a tetrahedral topology
+        if (fromModel.get()->getTopologyType() != sofa::geometry::ElementType::TETRAHEDRON)
+        {
+            msg_error() << "The type of the input topology '" << fromModel.getPath() << "' does not correspond to a tetrahedral topology.";
+            modelsOk = false;
+        }
+    }
 
+    // Check output topology
     if (!toModel)
     {
         // If the output topology link isn't set by the user, the TopologicalMapping::create method tries to find it.
         // If it is null at this point, it means no output mesh topology could be found.
         msg_error() << "No output mesh topology found. Consider setting the '" << toModel.getName() << "' data attribute.";
         modelsOk = false;
     }
-
-    // Making sure the output topology is derived from the triangle topology container
-    if (!dynamic_cast<container::dynamic::TriangleSetTopologyContainer *>(toModel.get())) {
-        msg_error() << "The output topology '" << toModel.getPath() << "' is not a derived class of TriangleSetTopologyContainer. "
-                    << "Consider setting the '" << toModel.getName() << "' data attribute to a valid"
-                                                                        " TriangleSetTopologyContainer derived object.";
-        modelsOk = false;
-    } else {
-        // Making sure a topology modifier exists at the same level as the output topology
-        container::dynamic::TriangleSetTopologyModifier *to_tstm;
-        toModel->getContext()->get(to_tstm);
-        if (!to_tstm) {
-            msg_error() << "No TriangleSetTopologyModifier found in the output topology node '"
-                        << toModel->getContext()->getName() << "'.";
+    else
+    {
+        // Making sure the output topology is derived from the triangle topology container
+        if (!dynamic_cast<container::dynamic::TriangleSetTopologyContainer *>(toModel.get()))
+        {
+            msg_error() << "The input topology '" << toModel.getPath() << "' is not homogeneous with a TriangleSetTopologyContainer. The '" << toModel.getName() << "' data attribute must be linked to a valid component, among the following list of eligible components:" << msgendl
+                                   << sofa::core::ObjectFactory::getInstance()->listClassesDerivedFrom<container::dynamic::TriangleSetTopologyContainer>();
```

Comment:
I tested what would be the error message. In this mapping case, it is: 
``` data attribute must be linked to a valid component, among the following list of eligible components: TetrahedronSetTopologyContainer, TriangleSetTopologyContainer```

which doesn't work, because putting a TetrahedronSetTopologyContainer as output will result in a strange behavior. Not sure if it will crash or if it will be like a "TopologicalIdentityMapping" 🤔 

I suggest to move the checks inside a commun method in TopologicalMapping and just test the topologyType of the containters